### PR TITLE
Rewrite dynamic api example

### PR DESCRIPTION
**Pull request by MikailBag**
This example now shows how one can use kube to inspect arbitrary
resources.

### CODE REVIEW

File: examples/dynamic_api.rs [hunk 1]
```diff
@@ -18,31 +24,50 @@ async fn main() -> anyhow::Result<()> {
...
+    println!("{}", group_version);
+    for ar in apis.resources {
+        if !ar.verbs.contains(&"list".to_string()) {
+            continue;
+        }
+        if group_version.starts_with("discovery.k8s.io/") && ar.kind == "EndpointSlice" ||
+          group_version.starts_with("metrics.k8s.io/")  {
+            eprintln!("\tFIXME: skipping kind which would be otherwise incorrectly pluralized");
+            continue;
+        }
```
Comment (clux) [inline examples/dynamic_api.rs:55]: Ah, this is what you are talking about in #467.
Didn't realise there were counter-examples now. Was hoping they stayed consistent :(

File: examples/dynamic_api.rs [hunk 2]
```diff
@@ -18,31 +24,50 @@ async fn main() -> anyhow::Result<()> {
...
+            let name = item.name();
+            let ns = item.namespace();
+            match ns {
+                Some(ns) => {
+                    println!("\t\t{}/{}", ns, name);
+                }
+                None => {
+                    println!("\t\t{}", name);
+                }
+            }
```
Comment (clux) [inline examples/dynamic_api.rs:70]: The `DynamicObject` implements `Meta` now, right. We can use `Meta::name` and `Meta::namespace` here for a one-liner?

**Review by clux — APPROVED** [review]
small nits. appreciate the quick pr!

File: examples/dynamic_api.rs [hunk 3]
```diff
@@ -18,31 +24,50 @@ async fn main() -> anyhow::Result<()> {
...
+            continue;
+        }
+        if group_version.starts_with("discovery.k8s.io/") && ar.kind == "EndpointSlice" ||
+          group_version.starts_with("metrics.k8s.io/")  {
+            eprintln!("\tFIXME: skipping kind which would be otherwise incorrectly pluralized");
+            continue;
+        }
+        println!("\t{}", ar.kind);
+        let api = DynamicResource::from_api_resource(&ar, &apis.group_version)
+            .into_api::<DynamicObject>(client.clone());
```
Comment (clux) [inline examples/dynamic_api.rs:58]: is anything here dealing with namespaces, or is it just cluster / all querying?

Comment (MikailBag) [reply examples/dynamic_api.rs:58]: We are listing objects from all namespaces.

I will add an option to only list resources from one namespace.

**Comment by clux** [comment]
Did a small cleanup to it in master to remove the special casing after we merged #470.
Also tweaked tracing instrumentation a bit because it looked a bit awkward with log (guessing that's why you were using println)

**Pull request merged**